cmd/gpu_fakedev/gpu_fakedev.go

```diff
@@ -289,6 +293,85 @@ func generateDriFiles(opts genOptions) {
 		}
 	}
 	log.Printf("Done, created %d dirs, %d devices and %d files.", opts.dirs, opts.devs, opts.files)
+
+	makeXelinkSideCar(opts)
+}
+
+func makeXelinkSideCar(opts genOptions) {
+	topology := opts.Capabilities["connection-topology"]
+	gpus := opts.DevCount
+	tiles := opts.TilesPerDev
+	connections := opts.Capabilities["connections"]
+
+	if topology != fullyConnected {
+		saveSideCarFile(connections)
+	} else {
+		saveSideCarFile(buildConnectionList(gpus, tiles))
+	}
+	log.Printf("XELINK: generate xelink sidecar label file, using (GPUs: %d, Tiles: %d, Topology: %s)", gpus, tiles, topology)
+}
+
+func buildConnectionList(gpus, tiles int) string {
+	var nodes = make([]string, 0)
+
+	for mm := 0; mm < gpus; mm++ {
+		for nn := 0; nn < tiles; nn++ {
+			nodes = append(nodes, fmt.Sprintf("%d.%d", mm, nn))
+		}
+	}
+
+	var links = make(map[string]struct{}, 0)
+
+	var smap = make([]string, 0)
+
+	for _, from := range nodes {
+		for _, to := range nodes {
+			// no self links, TODO ignore in-gpu xelinks
+			if to == from {
+				continue
+			}
+
+			link := fmt.Sprintf("%s-%s", to, from)
+
+			reverselink := fmt.Sprintf("%s-%s", from, to)
+			if _, exists := links[reverselink]; !exists {
+				links[link] = struct{}{}
```

Comment:
Looks IMHO slightly simpler with  bool instead of struct:
```suggestion
	var links = make(map[string]bool, 0)

	var smap = make([]string, 0)

	for _, from := range nodes {
		for _, to := range nodes {
			// no self links, TODO ignore in-gpu xelinks
			if to == from {
				continue
			}

			link := fmt.Sprintf("%s-%s", to, from)

			reverselink := fmt.Sprintf("%s-%s", from, to)
			if _, exists := links[reverselink]; !exists {
				links[link] = true
```

Reply:
That is now changed